### PR TITLE
New signal slots for Qt 6

### LINKED ISSUE #1993: Qt 5: convert connect calls to new generic syntax
https://wiki.qt.io/New_Signal_Slot_Syntax

### PR DESCRIPTION
**Pull request by dcwhite**
Convert almost all of the old-style SIGNAL/SLOT Qt macros to the new function pointer syntax. Nearly closes #1993.

Also includes some new GUI features.

Requesting lots of reviewers to test for GUI bugs (non-working buttons or other widgets, mostly).

I'll post a link to the binaries here once they're done.

### CODE REVIEW

**Comment by dcwhite** [comment]
Looks like I need to update some calls for older Qt on older Linux. Fix coming soon

File: src/Interface/Modules/Render/OsprayViewerDialog.cc [hunk 1]
```diff
@@ -104,25 +104,25 @@ OsprayViewerDialog::OsprayViewerDialog(const std::string& name, ModuleStateHandl
...
-  connect(configDialog_->cameraViewUpYDoubleSpinBox_, SIGNAL(valueChanged(double)), this, SLOT(setViewportCamera()));
-  connect(configDialog_->cameraViewUpZDoubleSpinBox_, SIGNAL(valueChanged(double)), this, SLOT(setViewportCamera()));
-
-  connect(configDialog_->ambientLightColorRDoubleSpinBox_, SIGNAL(valueChanged(double)), this, SLOT(setLightColor()));
-  connect(configDialog_->ambientLightColorGDoubleSpinBox_, SIGNAL(valueChanged(double)), this, SLOT(setLightColor()));
-  connect(configDialog_->ambientLightColorBDoubleSpinBox_, SIGNAL(valueChanged(double)), this, SLOT(setLightColor()));
-  connect(configDialog_->directionalLightColorRDoubleSpinBox_, SIGNAL(valueChanged(double)), this, SLOT(setLightColor()));
-  connect(configDialog_->directionalLightColorGDoubleSpinBox_, SIGNAL(valueChanged(double)), this, SLOT(setLightColor()));
-  connect(configDialog_->directionalLightColorBDoubleSpinBox_, SIGNAL(valueChanged(double)), this, SLOT(setLightColor()));
+  connect(configDialog_->viewerHeightSpinBox_, &QSpinBox::valueChanged, this, &OsprayViewerDialog::setHeight);
```
Comment (jessdtate) [inline src/Interface/Modules/Render/OsprayViewerDialog.cc:107]: build error:
`no matching member function for call to 'connect'`
and many similar errors following

Comment (dcwhite) [reply src/Interface/Modules/Render/OsprayViewerDialog.cc:107]: I didn't build with Ospray on so I will have to fix those too

**Review by jessdtate — APPROVED** [review]

**Comment by jessdtate** [comment]
@dcwhite make sure that you review changes to splitfieldbyconnectedregion

**Comment by jab0707** [comment]
are there binaries yet for testing?

**Comment by dcwhite** [comment]
https://github.com/SCIInstitute/SCIRun/suites/6181798863/artifacts/217399830

**Comment by dcwhite** [comment]
this branch does not have the packaging fix in it, so that's the non-python package

**Comment by jab0707** [comment]
does there also happen to be a windows version? If not I can just test mac.

**Comment by dcwhite** [comment]
https://github.com/SCIInstitute/SCIRun/suites/6181798864/artifacts/217356480

**Review by jab0707 — APPROVED** [review]

**Pull request merged**